### PR TITLE
new(modern_bpf): add `fcntl` and `shutdown` syscalls

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area driver-modern-bpf

/area libpman

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR is part of a series https://github.com/falcosecurity/libs/issues/513, the final aim is to support the most important syscalls also in the new probe. This PR introduces:

* `fcntl`
* `shutdown`

This syscall wasn't in the original set but probably it will be needed by the new `interesting` syscall approach

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(modern_bpf): add `fcntl` and `shutdown` syscall
```

### CODE REVIEW

**Review by FedeDP — DISMISSED** [review]
/approve

**Comment by Andreagit97** [comment]
/hold
This PR will contain at least also the `shutdown` syscall

**Comment by Andreagit97** [comment]
This should be ready I've tested it on `s390x` `aarch64` and `x86` :)

**Comment by Andreagit97** [comment]
Hey, @hbrueckner thank you very much for having tested it, as always! I think you have gained a great knowledge of this part of the codebase, would you like to become a reviewer of it? This is nothing different from what you are already doing... when you have some spare time you can review PRs like this, but being a reviewer you can also put the LGTM approval, in this way other maintainers can easily understand that the code is ok for you. I think that all other maintainers agree with me on this, it's quite clear the effort that you are putting into it, so I think you deserve it :) @leogr @LucaGuerra @FedeDP @Molter73

**Comment by leogr** [comment]
> Hey @hbrueckner thank you very much for having tested it, as always! I think you have gained a great knowledge of this part of the codebase, would you like to become a reviewer of it? This is nothing different from what you are already doing... when you have some spare time you can review PRs like this, but being a reviewer you can also put the LGTM approval, in this way other maintainers can easily understand that the code is ok for you. I think that all other maintainers agree with me on this, it's quite clear the effort that you are putting into it, so I think you deserve it :) @leogr @LucaGuerra @FedeDP @Molter73

big +1 from me :)

**Comment by hbrueckner** [comment]
Hi @Andreagit97,  @leogr, and @FedeDP,

> I fully agree with Andrea; @hbrueckner you are doing such a tremendous job in reviewing and testing every modern-bpf related PR, it's really stunning! Please take a moment and think about joining us :) it would be great to have such a domain expert helping us with code reviews!

I am delighted to join your team as reviewer! Many thanks for this offer and looking forward for the next reviews. Thanks a lot!

**Comment by leogr** [comment]
> Hi @Andreagit97, @leogr, and @FedeDP,
> 
> > I fully agree with Andrea; @hbrueckner you are doing such a tremendous job in reviewing and testing every modern-bpf related PR, it's really stunning! Please take a moment and think about joining us :) it would be great to have such a domain expert helping us with code reviews!
> 
> I am delighted to join your team as reviewer! Many thanks for this offer and looking forward for the next reviews. Thanks a lot!

:partying_face: 

Please open a PR to add yourself in https://github.com/falcosecurity/libs/blob/master/OWNERS under the reviewers' section, ie :point_down: 
```
approvers:
  - leogr
  - gnosek
  - mstemm
  - fededp
  - andreagit97
  - molter73
  - lucaguerra
reviewers:
  - hbrueckner
emeritus_approvers:
  - fntlnz
  - leodido
  - ldegio
```

**Comment by FedeDP** [comment]
That's huge @hbrueckner ! Thank you very much, it is a relief having you on board with the reviews :)

**Review by leogr — APPROVED** [review]
LGTM

I've not canceled the `hold` in case anyone else wants to double-check first.
Otherwise, @Andreagit97 feel free to unhold.

**Comment by Andreagit97** [comment]
Amazing news @hbrueckner :heart_eyes: !! If you need help feel free to ask, this is a PR that do it https://github.com/falcosecurity/libs/pull/290, you need to fill only the `reviewers` section :)

**Comment by Andreagit97** [comment]
I think this is ok :) 
/unhold

**Pull request merged**
